src/borg/archiver.py

```diff
@@ -2168,6 +2168,13 @@ def main():  # pragma: no cover
     except RemoteRepository.RPCError as e:
         msg = '%s\n%s' % (str(e), sysinfo())
         exit_code = EXIT_ERROR
+    except BrokenPipeError:
+        if args.func.__name__ in ['do_info', 'do_list', 'do_diff']:
```

Comment:
Yes, BrokenPipeError is ignored for known read-only commands by this approach.

For the second question, did you mean for which command was the pipe broken? If so, it was only for the info command; it worked fine for both list and diff. I intentionally included the two that worked just because of the comments in previous and current issues/PR related to this PR.